tests/concepts/test_table_contract.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from cohortextractor.backends.base import BaseBackend, Column, MappedTable
+from cohortextractor.concepts import tables
+from cohortextractor.concepts.table_contract import BackendContractError
+from cohortextractor.query_engines.base_sql import BaseSQLQueryEngine
+
+
+def test_basic_validation_that_table_implements_patients_contract():
+    with pytest.raises(BackendContractError):
+
+        class BadBackend(BaseBackend):
+            backend_id = "bad_test_backend"
+            query_engine_class = BaseSQLQueryEngine
+            patient_join_column = "Patient_ID"
+
+            patients = MappedTable(
+                implements=tables.Patients,
+                source="Patient",
+                columns=dict(
+                    date_of_birth=Column("date", source="DateOfBirth"),
+                ),
+            )
```

Comment:
I suppose that in an ideal world I'd prefer to see a happy-path test case as well.